spec/controllers/uploads_spec.rb

```diff
@@ -20,25 +20,68 @@
         and_return(user)
       subject { get :index }
     end
+  end
 
+  describe "#create" do
     it "uploads the file" do
       user = create(:user)
       allow_any_instance_of(ApplicationController).to receive(:current_user).
         and_return(user)
 
-      get :create, params: { audio: file_to_upload }
+      expect {
+        post :create, { audio: file_to_upload }
+      }.to change{ Upload.count }.by(1)
+
+      expect(response).to redirect_to(new_upload_path)
+      expect(flash[:error]).not_to be_present
+    end
+
+    it "doesn't upload a file for an non-authorized user" do
+      expect {
+        post :create, { audio: file_to_upload }
+      }.to change{ Upload.count }.by(0)
+
+      expect(response).to redirect_to(root_path)
+    end
+  end
+
+  describe "#destroy" do
+    it "deletes the upload and redirects to the new upload path" do
+      user = create(:user)
+      allow_any_instance_of(ApplicationController).to receive(:current_user).
+        and_return(user)
+      post :create, { audio: file_to_upload }
+
+      expect { delete :destroy, id: Upload.last
+        }.to change{ Upload.count }.by(-1)
```

Comment:
`end` at 56, 8 is not aligned with `expect { delete :destroy, id: Upload.last` at 55, 6<br>Space missing to the left of {.